### PR TITLE
Drop loop kwarg from async_timeout.timeout

### PR DESCRIPTION
**Pull request by bdraco**
- async_timeout 4.0.0 drops the loop= kwarg and will fail if its passed

### CODE REVIEW

**Comment by tech165** [comment]
Hi, would it be possible to merge this pull request ? It is filling the logs on restart for home assistant. 

Thanks

**Comment by camrun91** [comment]
@Cinntax any chance we could get this pulled in?

**Comment by Cinntax** [comment]
Apologies for holding everything up- @bdraco @tech165 @bbrendon @camrun91 . Let me get a new release of this on pypi so it can be put into HA.

**Comment by raspman** [comment]
@Cinntax Thanks for your help.  Do you think this issue is causing the periodic disconnects/re-connects?

**Comment by tech165** [comment]
@Cinntax Thanks, greatly appreciated :)

@raspman I did not have any connections problems on my side, just warnings in the logs.  So I'm guessing this won't solve your issues.

**Comment by raspman** [comment]
I'm not sure about that.  I can tell you that this warning comes at almost the exact same time as Envisalink disconnects and gives an error.  It appears based on this there may be a correlation.  The disconnect is very short (3 minutes and it's back up.) If not, is there a main author of the Envisalink integration that can look at the disconnect issues, please?

https://github.com/home-assistant/core/issues/62686)


021-12-23 00:34:53 ERROR (MainThread) [pyenvisalink.envisalink_base_client] The server closed the connection. Reconnecting...
2021-12-23 00:35:23 WARNING (MainThread) [homeassistant.helpers.frame] Detected code that called async_timeout.timeout with loop keyword argument. The loop keyword argument is deprecated and calls will fail after Home Assistant 2022.2. Please report this issue.
Stack (most recent call last):
File "/usr/local/lib/python3.9/runpy.py", line 197, in _run_module_as_main
return _run_code(code, main_globals, None,
File "/usr/local/lib/python3.9/runpy.py", line 87, in _run_code
exec(code, run_globals)
File "/usr/src/homeassistant/homeassistant/__main__.py", line 331, in <module>
sys.exit(main())
File "/usr/src/homeassistant/homeassistant/__main__.py", line 318, in main
exit_code = runner.run(runtime_conf)
File "/usr/src/homeassistant/homeassistant/runner.py", line 121, in run
return loop.run_until_complete(setup_and_run_hass(runtime_config))
File "/usr/local/lib/python3.9/asyncio/base_events.py", line 629, in run_until_complete
self.run_forever()
File "/usr/local/lib/python3.9/asyncio/base_events.py", line 596, in run_forever
self._run_once()
File "/usr/local/lib/python3.9/asyncio/base_events.py", line 1890, in _run_once
handle._run()
File "/usr/local/lib/python3.9/asyncio/events.py", line 80, in _run
self._context.run(self._callback, *self._args)
File "/usr/local/lib/python3.9/site-packages/pyenvisalink/envisalink_base_client.py", line 83, in reconnect
await self.connect()
File "/usr/local/lib/python3.9/site-packages/pyenvisalink/envisalink_base_client.py", line 61, in connect
async with async_timeout.timeout(self._alarmPanel.connection_timeout, loop=self._eventLoop):
File "/usr/src/homeassistant/homeassistant/async_timeout_backcompat.py", line 19, in timeout
report(
File "/usr/src/homeassistant/homeassistant/helpers/frame.py", line 74, in report
_LOGGER.warning(msg, stack_info=True)
Stack (most recent call last):
File "/usr/local/lib/python3.9/runpy.py", line 197, in _run_module_as_main
return _run_code(code, main_globals, None,
File "/usr/local/lib/python3.9/runpy.py", line 87, in _run_code
exec(code, run_globals)
File "/usr/src/homeassistant/homeassistant/__main__.py", line 331, in <module>
sys.exit(main())
File "/usr/src/homeassistant/homeassistant/__main__.py", line 318, in main
exit_code = runner.run(runtime_conf)
File "/usr/src/homeassistant/homeassistant/runner.py", line 121, in run
return loop.run_until_complete(setup_and_run_hass(runtime_config))
File "/usr/local/lib/python3.9/asyncio/base_events.py", line 629, in run_until_complete
self.run_forever()
File "/usr/local/lib/python3.9/asyncio/base_events.py", line 596, in run_forever
self._run_once()
File "/usr/local/lib/python3.9/asyncio/base_events.py", line 1890, in _run_once
handle._run()
File "/usr/local/lib/python3.9/asyncio/events.py", line 80, in _run
self._context.run(self._callback, *self._args)
File "/usr/local/lib/python3.9/site-packages/pyenvisalink/envisalink_base_client.py", line 83, in reconnect
await self.connect()
File "/usr/local/lib/python3.9/site-packages/pyenvisalink/envisalink_base_client.py", line 61, in connect
async with async_timeout.timeout(self._alarmPanel.connection_timeout, loop=self._eventLoop):
File "/usr/src/homeassistant/homeassistant/async_timeout_backcompat.py", line 19, in timeout
report(
File "/usr/src/homeassistant/homeassistant/helpers/frame.py", line 74, in report
_LOGGER.warning(msg, stack_info=True)

**Comment by Cinntax** [comment]
Just an FYI- I haven’t been able to push a new version to pypi yet- they seem to be having an outage. So I’ll keep an eye on it- and deploy version 4.2 ASAP.

**Comment by Cinntax** [comment]
Looks like everything is back up again- so version 4.2 has been released and can be used in HA!

**Comment by raspman** [comment]
How do you update this in HA?  Does this resolve the disconnect issue?  I just updated to 12.6 and still getting:

 Logger: homeassistant.helpers.frame
Source: helpers/frame.py:74
First occurred: 11:52:49 AM (1 occurrences)
Last logged: 11:52:49 AM
Detected code that called async_timeout.timeout with loop keyword argument. The loop keyword argument is deprecated and calls will fail after Home Assistant 2022.2. Please report this issue.

**Comment by raspman** [comment]
@tech165 I'm in agreement with you at this point.  I've tried HA 202.11, because at least one person reported no more errors with this version, and I'm still getting (errors) disconnects, even without the warning I believe the above is supposed to fix:  2021-12-23 00:35:23 WARNING (MainThread) [homeassistant.helpers.frame] Detected code that called async_timeout.timeout with loop keyword argument. The loop keyword argument is deprecated and calls will fail after Home Assistant 2022.2. Please report this issue.

 In other words, even without this warning, I'm still getting the disconnects. I was hopeful there would be a solution to this, but after experiencing this for about a month, I'm thinking Envisalink won't be working with HA the way I need it to, that is, reliably.

**Comment by alexruffell** [comment]
Adding that in case the fix has been implemented, I am actually still getting the error. I am running all the latest versions on Home Assistant

```
2022-01-13 13:21:35 WARNING (MainThread) [homeassistant.helpers.frame] Detected code that called async_timeout.timeout with loop keyword argument. The loop keyword argument is deprecated and calls will fail after Home Assistant 2022.2. Please report this issue.
Stack (most recent call last):
  File "/usr/local/lib/python3.9/runpy.py", line 197, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "/usr/local/lib/python3.9/runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "/usr/src/homeassistant/homeassistant/__main__.py", line 331, in <module>
    sys.exit(main())
  File "/usr/src/homeassistant/homeassistant/__main__.py", line 318, in main
    exit_code = runner.run(runtime_conf)
  File "/usr/src/homeassistant/homeassistant/runner.py", line 121, in run
    return loop.run_until_complete(setup_and_run_hass(runtime_config))
  File "/usr/local/lib/python3.9/asyncio/base_events.py", line 629, in run_until_complete
    self.run_forever()
  File "/usr/local/lib/python3.9/asyncio/base_events.py", line 596, in run_forever
    self._run_once()
  File "/usr/local/lib/python3.9/asyncio/base_events.py", line 1890, in _run_once
    handle._run()
  File "/usr/local/lib/python3.9/asyncio/events.py", line 80, in _run
    self._context.run(self._callback, *self._args)
  File "/usr/local/lib/python3.9/site-packages/pyenvisalink/envisalink_base_client.py", line 61, in connect
    async with async_timeout.timeout(self._alarmPanel.connection_timeout, loop=self._eventLoop):
  File "/usr/src/homeassistant/homeassistant/async_timeout_backcompat.py", line 19, in timeout
    report(
  File "/usr/src/homeassistant/homeassistant/helpers/frame.py", line 74, in report
    _LOGGER.warning(msg, stack_info=True)
Stack (most recent call last):
  File "/usr/local/lib/python3.9/runpy.py", line 197, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "/usr/local/lib/python3.9/runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "/usr/src/homeassistant/homeassistant/__main__.py", line 331, in <module>
    sys.exit(main())
  File "/usr/src/homeassistant/homeassistant/__main__.py", line 318, in main
    exit_code = runner.run(runtime_conf)
  File "/usr/src/homeassistant/homeassistant/runner.py", line 121, in run
    return loop.run_until_complete(setup_and_run_hass(runtime_config))
  File "/usr/local/lib/python3.9/asyncio/base_events.py", line 629, in run_until_complete
    self.run_forever()
  File "/usr/local/lib/python3.9/asyncio/base_events.py", line 596, in run_forever
    self._run_once()
  File "/usr/local/lib/python3.9/asyncio/base_events.py", line 1890, in _run_once
    handle._run()
  File "/usr/local/lib/python3.9/asyncio/events.py", line 80, in _run
    self._context.run(self._callback, *self._args)
  File "/usr/local/lib/python3.9/site-packages/pyenvisalink/envisalink_base_client.py", line 61, in connect
    async with async_timeout.timeout(self._alarmPanel.connection_timeout, loop=self._eventLoop):
  File "/usr/src/homeassistant/homeassistant/async_timeout_backcompat.py", line 19, in timeout
    report(
  File "/usr/src/homeassistant/homeassistant/helpers/frame.py", line 74, in report
    _LOGGER.warning(msg, stack_info=True)
```
